test/node/regressions/many-request-handlers.test.ts

```diff
@@ -1,58 +1,111 @@
+
+beforeAll(async () => {
+  await httpServer.listen()
+
+  // Create a large number of request handlers.
+  const restHandlers = new Array(100).fill(null).map((_, index) => {
+    return http.post(
+      httpServer.http.url(`/resource/${index}`),
+      async ({ request }) => {
+        const text = await request.text()
+        return HttpResponse.text(text + index.toString())
+      },
+    )
+  })
+
+  const graphqlHandlers = new Array(100).fill(null).map((_, index) => {
+    return graphql.query(`Get${index}`, () => {
+      return HttpResponse.json({ data: { index } })
+    })
+  })
 
-beforeAll(() => {
+  server = setupServer(...restHandlers, ...graphqlHandlers)
   server.listen()
   vi.spyOn(process.stderr, 'write')
 })
 
-afterAll(() => {
+afterAll(async () => {
   server.close()
   vi.restoreAllMocks()
+  await httpServer.close()
 })
 
 it('does not print a memory leak warning when having many request handlers', async () => {
-  const httpResponse = await fetch('https://example.com/resource/42', {
+  const httpResponse = await fetch(`${httpServer.http.url(`/resource/99`)}`, {
     method: 'POST',
     body: 'request-body-',
   }).then((response) => response.text())
+  expect(spy).toHaveBeenNthCalledWith(1, 200, expect.any(AbortSignal))
+  // Each clone is a new AbortSignal listener which needs to be registered
+  expect(requestCloneSpy).toHaveBeenCalledTimes(100)
+  expect(process.stderr.write).not.toHaveBeenCalled()
+  requestCloneSpy.mockClear()
 
-  const graphqlResponse = await fetch('https://example.com', {
+  const graphqlResponse = await fetch(httpServer.http.url('/graphql'), {
     method: 'POST',
     headers: {
       'Content-Type': 'application/json',
     },
     body: JSON.stringify({
-      query: `query Get42 { index }`,
+      query: `query Get99 { index }`,
     }),
   }).then((response) => response.json())
 
+  expect(spy).toHaveBeenNthCalledWith(2, 200, expect.any(AbortSignal))
+  // Each clone is a new AbortSignal listener which needs to be registered
+  expect(requestCloneSpy).toHaveBeenCalledTimes(300)
+  requestCloneSpy.mockClear()
+
   // Must not print any memory leak warnings.
-  expect(process.stderr.write).not.toHaveBeenCalled()
+  // expect(process.stderr.write).not.toHaveBeenCalled()
 
   // Must return the mocked response.
-  expect(httpResponse).toBe('request-body-42')
-  expect(graphqlResponse).toEqual({ data: { index: 42 } })
+  expect(httpResponse).toBe('request-body-99')
+  expect(graphqlResponse).toEqual({ data: { index: 99 } })
+  // Must not print any memory leak warnings.
+  expect(process.stderr.write).not.toHaveBeenCalled()
+
+  const unhandledResponse = await fetch(httpServer.http.url('/graphql'), {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      query: `query NotDefinedAtAll { index }`,
+    }),
+  })
+
+  expect(unhandledResponse.status).toEqual(500)
+
+  expect(requestCloneSpy).toHaveBeenCalledTimes(301)
```

Comment:
These requestClone spies are just to help showcase how many copies we're making, it's a lot!